service/frontend/operator_handler.go

```diff
@@ -318,13 +319,13 @@ func (h *OperatorHandlerImpl) addSearchAttributesSQL(
 				fmt.Sprintf(errTooManySearchAttributesMessage, cntUsed, saType.String()),
 			)
 		}
-		fieldToAliasMap[targetFieldName] = saName
+		upsertFieldToAliasMap[targetFieldName] = saName
```

Comment:
The `targetFieldName` is calculated based on the describe namespace response which can be stale when there're concurrent requests?

Reply:
Ugh. You're right. So, the only way to make it safe is to do all the logic inside `UpdateNamespace`, right?

Reply:
@yycptt Instead of moving the logic to namespace handler, I changed `UpdateNamespace` to check the operation is valid. That is, if the field name is already allocated, it will return an error. Thus, where there are concurrent requests, one of them might fail if the same field name is assigned to the alias.

Moving the logic to namespace handler will be a larger refactor, and will have a introduce a weird format for adding/removing aliases.

Reply:
👍 

My only concern is now this `CustomSearchAttributeAliases` field behaves differently than other fields for UpdateNamespace which could be confusing. We probably need to do some larger refactor later. 

Hmmm, for now, can you add some comment explaining this behavior of merging instead of overwriting for `CustomSearchAttributeAliases`?

Reply:
It's not only custom search attributes that has this behavior of merging. Bad binaries also do merging (L510 in `namespace_handler.go`).